dvc/remote/local.py

```diff
@@ -138,7 +139,9 @@ def getsize(path_info):
         return os.path.getsize(fspath_py35(path_info))
 
     def walk_files(self, path_info):
-        for fname in walk_files(path_info, self.repo.dvcignore):
+        assert isinstance(self.repo.tree, CleanTree)
```

Comment:
If `repo.tree` is `CleanTree(GitTree(...))` then this will pass, but it shouldn't. This should not test by cleanness but that we don't walk the git branch. `RemoteLocal` should only work with real dirs and files.